plugins/chat/assets/javascripts/discourse/models/chat-channel.js

```diff
@@ -91,11 +91,12 @@ export default class ChatChannel {
   @tracked archive;
   @tracked tracking;
   @tracked threadingEnabled = false;
-  @tracked threadTrackingOverview = new TrackedArray();
 
   threadsManager = new ChatThreadsManager(getOwner(this));
   messagesManager = new ChatMessagesManager(getOwner(this));
 
+  @tracked _threadTrackingOverview = new TrackedArray();
```

Comment:
How about we use a TrackedSet for this? this way we can do add/remove/has without having to user filter and having to worry about duplicates